andela/mmdp_site
# MMDP-512 Fix inability to delete user

**What does this PR do?**

Fixes inability to delete any user.

**Description of Task to be completed?**

All attempts to delete a user are rejected with 403 - You are not authorized to delete your own account.

**How should this be manually tested?**

1. Fetch all branches - `git fetch --all`.
2. Checkout to this branch - `git checkout bg-fix-user-delete-MMDP-512`.
3. Run the application - `yarn dev-server`.
4. Open the application and log in with an account with users delete permission.
5. Go to Users and ensure that you have existing users. If not add one.
5. Select the user and try deleting them. You should be able to delete them successfully.
6. Copy `userToken` from the browser local storage and use it on Postman as the `Authorization header` of a request in the format `Bearer <token>`.
6. Try to delete the currently logged user by making a request of the following format. `DELETE http://0.0.0.0:3000/api/v1/users/<username>`. You should a `403` with the message, `You are not authorized to delete your own account`.

**Any background context you want to provide?**
None

**What are the relevant Jira issues?**

[MMDP-512](https://viisaus.atlassian.net/browse/MMDP-512)

**Screenshots (if appropriate)**
Deleting user account
![Screenshot 2019-03-11 at 17 54 57](https://user-images.githubusercontent.com/8180548/54133308-e3a1f980-4426-11e9-8db6-f7eaeaae7ade.png)

Attempt to delete own account
![Screenshot 2019-03-11 at 17 53 42](https://user-images.githubusercontent.com/8180548/54133355-fae0e700-4426-11e9-8430-32efb5b9846b.png)

**Questions:**
None
